scripts/_functions_sass.R

```diff
@@ -0,0 +1,438 @@
+#' Write a Sass layer file used to split the final theme into four layers
+#' using the same logic as `sass::sass_layer_file()` and Quarto.
+#' @param dir The directory where the `_sass_layer_markers.css` file will be written.
+#' @return The path to the newly created Sass layer file.
+write_sass_layer_markers <- function(dir) {
+  sass_markers <- "/*-- scss:functions --*/
+// SPLIT: functions
+
+/*-- scss:defaults --*/
+// SPLIT: defaults
+
+/*-- scss:rules --*/
+// SPLIT: rules
+
+/*-- scss:mixins --*/
+// SPLIT: mixins
+"
+
+  path_sass_markers <- path(dir, "_sass_layer_markers.scss")
+  writeLines(sass_markers, path_sass_markers)
+
+  path_sass_markers
+}
+
+#' Create a base Bootstrap theme via {bslib}.
+bslib_bs_theme <- function(version, preset = "shiny") {
+  theme <- bs_theme(version, preset = preset)
+  # The bs3compat layer is for legacy markup only and isn't needed in py-shiny
+  bs_remove(theme, "bs3compat")
+}
+
+#' Create a full Shiny theme bundle with all dependencies.
+#'
+#' Adds styles for bslib components, shiny styles, `input_selectize()`, `input_slider()`
+#' and `input_date_range()`.
+#' @return A `bslib::bs_bundle()` object.
+bs_full_theme <- function(theme, path_sass_markers) {
+  bs_version <- bslib::theme_version(theme)
+
+  bs_bundle(
+    theme,
+    bslib = bslib_component_sass(),
+    shiny = shiny_sass(bs_version),
+    selectize = shiny_sass_selectize(bs_version),
+    ionrangeslider = shiny_sass_ionrangeslider(),
+    daterange = shiny_sass_daterange_picker(),
+    marker = sass_layer_file(path_sass_markers),
+  )
+}
+
+bslib_component_sass <- function() {
+  bslib:::component_dependency_sass_files()
+}
+
+shiny_sass <- function(bs_version) {
+  shiny:::shinyDependencySass(bs_version)
+}
+
+shiny_sass_selectize <- function(bs_version) {
+  shiny:::selectizeSass(bs_version)
+}
+
+shiny_sass_ionrangeslider <- function() {
+  sass::sass_layer(rules = shiny:::ionRangeSliderDependencySass())
```

Comment:
Is there a reason why this is in a layer, but the others aren't?

Reply:
To make sure it's treated as rules only. Maybe not necessary? I'm following what happens in `sass_partial()`.

Reply:
The ionrangeslide sass returns

```r
list(
  list(
    accent = "$component-active-bg"),
    sass::sass_file(system_file(package = "shiny", "www/shared/ionrangeslider/scss/shiny.scss")
  )
)
```

and this somehow translates to `$accent` being declared in the mixins portion of the Sass file, which is too early. This isn't a problem we've encountered on the R side because `sass_partial()` forces everything into rules.